docs/pbc/all/order-management-system/202311.0/base-shop/datapayload-conversion/state-machine/common-pitfalls-in-oms-design.md

```diff
@@ -171,3 +171,12 @@ Event does not appear as manual unless the previous command execution fails with
 Keeping both `onEnter` and `manual` commands can only be used for backup for the failed automated execution of the `onEnter` command with a manual event.
 
 {% endinfo_block %}
+
+## Calling OMS processing commands within a custom DB transaction
+
+**Issue:** On the project you decided to enclose complex processing, including OMS processing, in the transaction.
+
+OMS processing uses order items level lock to prevent processing of the same items in parallel. The lock information is stored as an entry in the table *spy_oms_state_machine_lock*.
```

Comment:
I think OOTB we are not providing parallel processing capabilities in the OMS.

Reply:
it may become so, when transaction is running for too long? So next job may pick up the same items..

Reply:
we do, @devromans. And this section is about a custom DB transaction, so not related to the OOTB code.

Reply:
@devromans  https://docs.spryker.com/docs/pbc/all/order-management-system/202311.0/base-shop/datapayload-conversion/state-machine/order-management-system-multi-thread.html